client/src/cmdlfem4x50.c

```diff
@@ -0,0 +1,720 @@
+    char pstring[100] = {0};
+
+    if (!success) {
+        
+        sprintf(pstring, "\n  writing new password " _RED_("failed"));
+        strcat(string, pstring);
+
+    } else {
+        
+        sprintf(pstring, "\n  writing new password " _GREEN_("successful"));
+        strcat(string, pstring);
+    }
+
+    PrintAndLogEx(NORMAL,"%s\n", string);
+}
+
+int CmdEM4x50WritePassword(const char *Cmd) {
+
+    // envokes changing the password of EM4x50 tag
+
+    bool errors = false, bpwd = false, bnewpwd = false;
+    char password[20] = {0x00}, new_password[20] = {0x00}, tmpbuff[3];
+    uint8_t cmdp = 0;
+    em4x50_data_t etd;
+    PacketResponseNG resp;
+
+    // init
+    etd.pwd_given = false;
+    etd.newpwd_given = false;
+
+    while (param_getchar(Cmd, cmdp) != 0x00 && !errors) {
+
+        switch (tolower(param_getchar(Cmd, cmdp))) {
+            case 'h':
+                return usage_lf_em4x50_write_password();
+
+            case 'p':
+                param_getstr(Cmd, cmdp + 1, password, sizeof(password));
+              
+                // validation
+                if (strlen(password) != 8) {
+                    PrintAndLogEx(WARNING, "\n  error, passwords has to be 4 bytes\n");
+                    errors = true;
+                    break;
+                } else {
+                    bpwd = true;
+                }
+              
+                // prepare given password
+                etd.pwd_given = true;
+                for (int i = 0; i < 4; i++)
+                    sscanf(strncpy(tmpbuff, password+2*i, 2), "%2hhx", &etd.password[i]);
+                  
+                cmdp += 2;
+                break;
+
+            case 'n':
+                param_getstr(Cmd, cmdp + 1, new_password, sizeof(new_password));
+              
+                // validation
+                if (strlen(new_password) != 8) {
+                    PrintAndLogEx(WARNING, "\n  error, passwords have to be 4 bytes\n");
+                    errors = true;
+                    break;
+                } else {
+                    bnewpwd = true;
+                }
+              
+                // prepare given password
+                etd.newpwd_given = true;
+                for (int i = 0; i < 4; i++)
+                    sscanf(strncpy(tmpbuff, new_password+2*i, 2), "%2hhx", &etd.new_password[i]);
+                  
+                cmdp += 2;
+                break;
+
+            default:
+                PrintAndLogEx(WARNING, "\n  Unknown parameter '%c'\n", param_getchar(Cmd, cmdp));
+                errors = true;
+                break;
+        }
+    }
+
+    if (errors || !bpwd || !bnewpwd)
+         return usage_lf_em4x50_write_password();
+
+    clearCommandBuffer();
+    SendCommandMIX(CMD_LF_EM4X50_WRITE_PASSWORD, 0, 0, 0, &etd, sizeof(etd));
+
+    if (!WaitForResponse(CMD_ACK, &resp)) {
+        PrintAndLogEx(WARNING, "\n  timeout while waiting for reply.\n");
+        return PM3_ESOFT;
```

Comment:
return code for timeouts,  PM3_ETIMEOUT   ;)

Reply:
done